doc/stdlib/strings.rst

```diff
@@ -219,151 +234,253 @@
+
+.. function:: uppercase(s::AbstractString)
 
    .. Docstring generated from Julia source
 
-   Returns ``string`` with all characters converted to uppercase.
+   Returns ``s`` with all characters converted to uppercase.
 
-.. function:: lowercase(string)
+   .. doctest::
+
+       julia> uppercase("Julia")
+       "JULIA"
+
+.. function:: lowercase(s::AbstractString)
 
    .. Docstring generated from Julia source
 
-   Returns ``string`` with all characters converted to lowercase.
+   Returns ``s`` with all characters converted to lowercase.
+
+   .. doctest::
+
+       julia> lowercase("STRINGS AND THINGS")
+       "strings and things"
 
-.. function:: ucfirst(string)
+.. function:: ucfirst(s::AbstractString)
 
    .. Docstring generated from Julia source
 
    Returns ``string`` with the first character converted to uppercase.
 
-.. function:: lcfirst(string)
+   .. doctest::
+
+       julia> ucfirst("python")
+       "Python"
+
+.. function:: lcfirst(s::AbstractString)
 
    .. Docstring generated from Julia source
 
    Returns ``string`` with the first character converted to lowercase.
 
-.. function:: join(strings, delim, [last])
+   .. doctest::
+
+       julia> lcfirst("Julia")
+       "julia"
+
+.. function:: join(io::IO, strings, delim, [last])
 
    .. Docstring generated from Julia source
 
-   Join an array of ``strings`` into a single string, inserting the given delimiter between adjacent strings. If ``last`` is given, it will be used instead of ``delim`` between the last two strings. For example
+   Join an array of ``strings`` into a single string, inserting the given delimiter between adjacent strings. If ``last`` is given, it will be used instead of ``delim`` between the last two strings. For example,
 
-   .. code-block:: julia
+   .. doctest::
 
-       join(["apples", "bananas", "pineapples"], ", ", " and ") == "apples, bananas and pineapples"
+       julia> join(["apples", "bananas", "pineapples"], ", ", " and ")
+       "apples, bananas and pineapples"
 
-   ``strings`` can be any iterable over elements ``x`` which are convertible to strings via ``print(io::IOBuffer, x)``\ .
+   ``strings`` can be any iterable over elements ``x`` which are convertible to strings via ``print(io::IOBuffer, x)``\ . ``strings`` will be printed to ``io``\ .
 
-.. function:: chop(string)
+.. function:: chop(s::AbstractString)
 
    .. Docstring generated from Julia source
 
    Remove the last character from a string.
 
-.. function:: chomp(string)
+   .. doctest::
+
+       julia> a = string("March")
+       "March"
+
+       julia> chop(a)
+       "Marc"
+
+.. function:: chomp(s::AbstractString)
 
    .. Docstring generated from Julia source
 
    Remove a single trailing newline from a string.
 
-.. function:: ind2chr(string, i)
+.. function:: ind2chr(s::AbstractString, i::Integer)
 
    .. Docstring generated from Julia source
 
-   Convert a byte index to a character index.
+   Convert a byte index ``i`` to a character index.
```

Comment:
If `s` isn't a bytestring, you need this, right @StefanKarpinski? What's the best way to phrase this?